solidity/facades/shape.js

```diff
@@ -20,6 +20,24 @@ Shape.fromJSON = function(json) {
 	// TODO: what else?
 	return shape;
 };
+Shape.address = function() {
+	return this.id;
+};
+Shape.owner = async function() {
+	return await this.contractPromise.methods.owner().call();
```

Comment:
Since `contractPromise` is a Promise containing the contract, these need to be like so:
```js
const contract = await this.contractPromise;
return await contract.methods.owner().call();
```
Of course, it wouldn't be too hard to change it so that the contract itself is stored, not a promise for it. But then you'd have to be more careful to initialize it with a web3 first.